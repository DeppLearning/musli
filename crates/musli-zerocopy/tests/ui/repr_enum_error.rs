use musli_zerocopy::ZeroCopy;

#[derive(ZeroCopy)]
#[repr(C, packed)]
enum ReprPackedC {
    Variant(u32),
}

#[derive(ZeroCopy)]
#[repr(u8, packed)]
enum ReprPackedU8 {
    Variant(u32),
}

#[derive(ZeroCopy)]
#[repr(C)]
enum ReprC {
    Variant(u32),
}

fn main() {
}
