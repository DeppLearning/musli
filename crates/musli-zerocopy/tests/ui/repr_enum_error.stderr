error: ZeroCopy: repr(packed) is only supported on structs
 --> tests/ui/repr_enum_error.rs:4:17
  |
4 | #[repr(C, packed)]
  |                 ^

error: ZeroCopy: repr(packed) is only supported on structs
  --> tests/ui/repr_enum_error.rs:10:18
   |
10 | #[repr(u8, packed)]
   |                  ^

error: ZeroCopy: only supported for repr(i*) or repr(u*) enums
  --> tests/ui/repr_enum_error.rs:16:8
   |
16 | #[repr(C)]
   |        ^

error[E0517]: attribute should be applied to a struct or union
 --> tests/ui/repr_enum_error.rs:4:11
  |
4 |   #[repr(C, packed)]
  |             ^^^^^^
5 | / enum ReprPackedC {
6 | |     Variant(u32),
7 | | }
  | |_- not a struct or union

error[E0517]: attribute should be applied to a struct or union
  --> tests/ui/repr_enum_error.rs:10:12
   |
10 |   #[repr(u8, packed)]
   |              ^^^^^^
11 | / enum ReprPackedU8 {
12 | |     Variant(u32),
13 | | }
   | |_- not a struct or union
